src/components/Activity.tsx

```diff
@@ -1,15 +1,24 @@
 import ActivityCard from './ActivityCard'
 import { useAllActivities } from '../hooks/useAllActivities';
+import { CircularProgress } from '@mui/material';
 
 
 const Activity: React.FC = () => {
-  const allActivities = useAllActivities().data;
+  const allActivities = useAllActivities().apiResponse.data;
+  const loading = useAllActivities().loading;
+  console.log(loading)
+
+  const loadingStyle = {
+    display: 'flex',
+    mr: 'auto',
+    ml: 'auto' 
+  }
 
   return (
     <>
-      {allActivities.map((item) =>
-        <ActivityCard activityItem={item} key={item.activity_id} />
-      )}
+      { loading || allActivities === undefined ?
```

Comment:
What is the thought behind 'or' || ?

Reply:
Thought I had to specify if allActivities are undefined,
but changed it to: '{ isLoading ?'